### PR TITLE
s3prl upstream for mos prediction

### PR DESCRIPTION
**Pull request by RayTzeng**
upstream implementation of "Utilizing Self-supervised Representations for MOS Prediction"

### CODE REVIEW

**Comment by leo19941227** [comment]
Hey @RayTzeng !

Thanks for the great work. Do you plan to include the downstream finetuning parts?

**Comment by RayTzeng** [comment]
I can also include the downstream finetuning parts, should I open a new pull request?

**Comment by leo19941227** [comment]
SGTM. I will merge this PR after some testing.

File: upstream/mos_prediction/expert.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+        return {"scores": scores}
+
+    def preprocessor(self, wav):
+
+        wav_segments = unfold_segments(wav, self.segments_durations)
+        return wav_segments
+
+    def _get_mos_upstream(self):
+        mos_upstream = torch.hub.load('s3prl/s3prl', self.upstream_type)
```
Comment (leo19941227) [inline upstream/mos_prediction/expert.py:84]: This way of calling upstream will use `torch.hub` to fetch the one located on the **origin/master**, which will not reflect the local change. After I package the s3prl, now there is a prefered way of using upstream models.

```
import s3prl.hub as hub
mos_upstream = getattr(hub, self.upstream_type)
```

**Review by leo19941227 — COMMENTED** [review]
I have just one minor suggestion.
Please fetch the lastest master and try to merge the master into this mos branch.

File: upstream/mos_prediction/expert.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        flattened_wavs_segments = [
+            wav_segment
+            for wav_segments in wavs_segments
+            for wav_segment in wav_segments
+        ]
+        wav_segments_lengths = [len(wav_segments) for wav_segments in wavs_segments]
+        prefix_sums = list(accumulate(wav_segments_lengths, initial=0))
+
+        
+        features = self.mos_upstream(flattened_wavs_segments)
```
Comment (leo19941227) [inline upstream/mos_prediction/expert.py:62]: You might need to update the upstream usage.
Originally we input a list of waveform and output a list of representation.
Now we change to output a dictionary, whose `last_hidden_state` key correspond to the original representation you used.
From this point of view, I think the current state of this PR will not work on the latest master. You might need to first merge the master into your current branch and test it work.

**Pull request closed without merging**